src/librustc/mir/mod.rs

```diff
@@ -1515,8 +1515,10 @@ pub enum AggregateKind<'tcx> {
     Array(Ty<'tcx>),
     Tuple,
 
-    /// The second field is variant number (discriminant), it's equal
-    /// to 0 for struct and union expressions. The fourth field is
+    /// The second field is variant number (discriminant),
+    /// i.e. the index of the variant within AdtDef::variants.
+    ///
+    /// It's equal to 0 for struct and union expressions. The fourth field is
```

Comment:
The newline seems misplaced. Also, we should avoid using "discriminant", because *that's a different thing that may not be equal to the variant index*. All the relevant code uses "variant index" (not "variant number").